src/Analyzers/CSharp/CodeFixes/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.cs

```diff
@@ -97,8 +98,12 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 if (shouldRemoveNextStatement)
                 {
                     // Already morphed into the top-level switch expression.
-                    var nextStatement = switchStatement.GetNextStatement();
+                    SyntaxNode nextStatement = switchStatement.GetNextStatement();
                     Debug.Assert(nextStatement.IsKind(SyntaxKind.ThrowStatement, SyntaxKind.ReturnStatement));
+                    if (nextStatement.IsParentKind(SyntaxKind.GlobalStatement))
+                    {
+                        nextStatement = nextStatement.Parent;
+                    }
                     editor.RemoveNode(nextStatement);
```

Comment:
```suggestion
                    editor.RemoveNode(nextStatement.IsParentKind(SyntaxKind.GlobalStatement) ? nextStatement.Parent : nextStatement);
```

(can keep teh var above as well).